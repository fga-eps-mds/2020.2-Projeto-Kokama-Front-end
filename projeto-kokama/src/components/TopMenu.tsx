import React from "react";
import {
  Platform,
  StatusBar,
  StyleSheet,
  View,
  Image,
  Text,
} from "react-native";
import Colors from "../assets/Colors";

const STATUSBAR_HEIGHT = Platform.OS === "ios" ? 20 : StatusBar.currentHeight;

const styles = StyleSheet.create({
  logoArea: {
    flexDirection: "column",
    marginTop: STATUSBAR_HEIGHT,
    paddingTop: 5,
    borderBottomWidth: 1,
    borderColor: Colors.DARK_GRAY,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: Colors.LIGHT_GRAY,
  },
  logo: {
    borderRadius: 100,
    height: 50,
    width: 50,
  },
  windowName: {
    fontSize: 20,
    marginBottom: 5,
  },
});

interface Props {
  name: string;
}

const TopMenu = (props: Props) => {
  return (
    <View>
      <StatusBar translucent backgroundColor={Colors.RED} />
      <View style={styles.logoArea}>
        <Image style={styles.logo} source={require("../assets/img/logo.png")} />
        <Text style={styles.windowName}>{props.name}</Text>
      </View>
    </View>
  );
};

export default TopMenu;
