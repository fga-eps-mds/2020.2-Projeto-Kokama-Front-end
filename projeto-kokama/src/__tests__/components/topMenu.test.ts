import renderer from 'react-test-renderer';
import TopMenu from '../../components/TopMenu'


test("Check TopMenu render", () => {

    let menu = TopMenu({name:'teste'})
    const tree = renderer.create(menu)
    expect(tree).toMatchSnapshot();
})