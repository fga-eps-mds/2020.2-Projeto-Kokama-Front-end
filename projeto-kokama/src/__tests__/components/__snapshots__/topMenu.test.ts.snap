// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check TopMenu render 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#E4DFDF",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 100,
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginBottom": 35,
        }
      }
    >
      teste
    </Text>
  </View>
</View>
`;
